### ISSUE TITLE
Ability to evaluate expressions and functions

### ISSUE THREAD

**Issue by RedEyedDog** [meddbase/isotope80#11]
Hi i'm looking to replicate this functionality that puppeteersharp uses:

```
      var jsCode2 = $"async () => await new Promise((resolve) => {{SM.Login('{username}', '{key}', (r) =>{{if (r && r.success){{resolve(r);}}}}); }})";

            await this
                .page
                .EvaluateFunctionAsync<object>(jsCode2);

            var jsCode3 = @"SM.Player";

            return await page
                .EvaluateExpressionAsync<LoggedInUser>(jsCode3);
```
It has the ability to evaluate Expressions and Functions is that possible?

**Comment by louthy** [comment]
@RedEyedDog Hi Christopher, I'm not example sure about the functionality of PuppeteerSharp, could you explain the use-case and requirement?

**Comment by RedEyedDog** [comment]
Hi thanks ill check that out. I think there is a problem with the readme. Is this correct:
`var logic =  nav("https://twitter.com/login")
              .Bind(x => className("js-username-field"), "Your Email Address"));
`

too many brackets? i thinking its missing sendKeys. I can raise a PR if that helps?

**Comment by sortbyfirstname** [comment]
@RedEyedDog Hi Christopher, I thought I'd correct the error you found in the README quickly as I was already working on a PR. This has been corrected in commit c2f1284, it was just missing `sendKeys(` as you said - well spotted.

**Comment by RedEyedDog** [comment]
Ok thanks guys
